src/System.Net.Sockets/tests/FunctionalTests/UnixDomainSocketTest.netcoreapp.cs

```diff
@@ -443,5 +442,29 @@ private static string GetRandomNonExistingFilePath()
 
             return result;
         }
+
+        private static bool PlatformDoesntSupportUnixDomainSockets => s_platformSupportsUnixDomainSockets.Value;
+
+        private static bool PlatformSupportsUnixDomainSockets => s_platformSupportsUnixDomainSockets.Value;
+
+        private static readonly Lazy<bool> s_platformSupportsUnixDomainSockets = new Lazy<bool>(() =>
+        {
+            // All Unixes should support UDS.
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return true;
+            }
+
+            // Some Windows will.
+            try
+            {
+                new Socket(AddressFamily.Unix, SocketType.Stream, ProtocolType.Unspecified).Dispose();
```

Comment:
This assumes that our product code is doing the right thing.  Would it be better to P/Invoke into Windows directly to see if it is supported?

Reply:
To check whether a given socket domain, type or protocol combo is supported, the best thing is to call the `socket` routine and check the return value.

Reply:
> To check whether a given socket domain, type or protocol combo is supported, the best thing is to call the socket routine and check the return value.

That's essentially what this code is doing, just via the .NET Socket type.

But to Eric's point, sure, I can just P/Invoke directly.